apps/arena/lib/arena/game/effect.ex

```diff
@@ -177,67 +187,68 @@ defmodule Arena.Game.Effect do
           now - Map.get(mechanic_params, :last_application_time) >= mechanic_params.effect_delay_ms
 
       if execute_multiple_times? and enough_time_passed? do
-        do_effect_mechanics(game_state, player, effect, mechanic)
+        do_effect_mechanics(game_state, entity, effect, mechanic)
         |> put_in_effect(effect, [:effect_mechanics, mechanic_name, :last_application_time], now)
       else
-        player
+        entity
       end
     end)
   end
 
-  defp do_effect_mechanics(game_state, player, effect, {:pull, pull_params}) do
+  defp do_effect_mechanics(game_state, entity, effect, {:pull, pull_params}) do
     case Map.get(game_state.pools, effect.owner_id) do
       nil ->
-        player
+        entity
 
-      %{position: pool_position} when pool_position == player.position ->
-        player
+      %{position: pool_position} when pool_position == entity.position ->
+        entity
 
       pool ->
-        if player.aditional_info.damage_immunity do
-          player
+        if entity.aditional_info.damage_immunity do
```

Comment:
```suggestion
        if Map. has_key?(entity.aditional_info, :damage_immunity) do
```

If the entity doesn't have that key, the game crashes